tests/jobs/misfit_preprocessor/unit/test_config.py

```diff
@@ -372,3 +389,54 @@ def test_default_observations(observation_keys):
 
     assert config.valid, config.errors
     assert sorted(observation_keys) == sorted(config.snapshot.observations)
+
+
+def test_new_config():
+    config_data = {
+        "observations": ["WWPR"],
+        "workflow": {
+            "auto_scale": {
+                "clustering": {
+                    "hierarchical": {"method": "complete", "metric": "jensenshannon"}
+                },
+                "pca": {"threshold": 0.98},
+            }
+        },
+    }
+    config = misfit_preprocessor.config.MisfitPreprocessorConfig(config_data, ["WWPR"])
+    assert config.valid, config.errors
+    assert config.snapshot.workflow.method == "auto_scale"
+
+
+def test_config_workflow():
+    config_data = {
+        "observations": ["WWPR"],
+        "workflow": {
+            "spearman_correlation": {
+                "clustering": {
+                    "hierarchical": {"method": "complete", "metric": "jensenshannon"}
+                },
+                "pca": {"threshold": 0.98},
+            }
+        },
+    }
+    config = misfit_preprocessor.config.MisfitPreprocessorConfig(config_data, ["WWPR"])
+    assert config.valid, config.errors
+    assert config.snapshot.workflow.method == "spearman_correlation"
+
+
+@pytest.mark.parametrize("workflow", [{"spearman_correlation": {}}, {"auto_scale": {}}])
+def test_config_workflow_valid(workflow):
+    config_data = {"workflow": workflow}
+    config = misfit_preprocessor.config.MisfitPreprocessorConfig(config_data, ["WWPR"])
+    assert config.valid, config.errors
+    assert config.snapshot.workflow.method == list(workflow.keys())[0]
+
+
+@pytest.mark.parametrize("workflow", [{"spearman_correlation": {}, "auto_scale": {}}])
+def test_config_workflow_invalid(workflow):
+    config_data = {"workflow": workflow}
+    #    with pytest.raises(ValueError):
+    config = misfit_preprocessor.config.MisfitPreprocessorConfig(config_data, ["WWPR"])
+
+    assert not config.valid
```

Comment:
Perhaps also check that the error message is what we would expect?

Reply:
The error message in config.errors is empty for this valid config.
Maybe we could try to check the error message in `test_config_workflow_invalid`?

Reply:
We made a test for that!